src/lib/homestore_backend/tests/hs_blob_tests.cpp

```diff
@@ -139,71 +139,148 @@ TEST_F(HomeObjectFixture, BasicPutGetDelBlobWRestart) {
+        pg_shard_id_vec[1].emplace_back(shard.id);
+        pg_blob_id[i] = 0;
+        LOGINFO("pg {} shard {}", pg_id, shard.id);
+    }
+    put_blobs(pg_shard_id_vec, num_blobs_per_shard, pg_blob_id);
+
+    PG* pg;
+    {
+        auto lg = std::shared_lock(_obj_inst->_pg_lock);
+        auto iter = _obj_inst->_pg_map.find(pg_id);
+        ASSERT_TRUE(iter != _obj_inst->_pg_map.end());
+        pg = iter->second.get();
+    }
+
+    //Construct shards as [sealed, open, filtered]
+    seal_shard(pg->shards_.front()->info.id);
+    ASSERT_EQ(pg->shards_.front()->info.state, homeobject::ShardInfo::State::SEALED);
+    //Filter out the last shard
+    auto snp_lsn = pg->shards_.back()->info.lsn - 1;
+    //Delete some blobs: delete the first blob of each shard
+    blob_id_t current_blob_id{0};
+    for (auto& shard : pg->shards_) {
+        del_blob(pg->pg_info_.id, shard->info.id, current_blob_id);
+        current_blob_id += num_blobs_per_shard;
+    }
+
+    auto pg_iter =
+        std::make_shared< HSHomeObject::PGBlobIterator >(*_obj_inst, pg->pg_info_.replica_set_uuid, snp_lsn);
+    ASSERT_EQ(pg_iter->shard_list_.size(), num_shards_per_pg - 1);
+    // Created blob sizes are distributed in range (1, 16kb)
+    pg_iter->max_batch_size_ = 16 * 1024;
+
+    // Verify PG meta data
+    sisl::io_blob_safe meta_blob;
+    pg_iter->create_pg_snapshot_data(meta_blob);
+    ASSERT_TRUE(meta_blob.size() > 0);
+
+    SyncMessageHeader* header = r_cast< SyncMessageHeader* >(meta_blob.bytes());
+    ASSERT_EQ(header->msg_type, SyncMessageType::PG_META);
+    auto pg_msg = GetSizePrefixedResyncPGMetaData(meta_blob.cbytes() + sizeof(SyncMessageHeader));
+    ASSERT_EQ(pg_msg->pg_id(), pg->pg_info_.id);
+    auto u1 = pg_msg->replica_set_uuid();
+    auto u2 = pg->pg_info_.replica_set_uuid;
+    ASSERT_EQ(std::string(u1->begin(), u1->end()), std::string(u2.begin(), u2.end()));
+    ASSERT_EQ(pg_msg->blob_seq_num(), pg->durable_entities().blob_sequence_num.load());
+    ASSERT_EQ(pg_msg->shard_seq_num(), pg->shard_sequence_num_);
+
+    auto msg_members = pg_msg->members();
+    ASSERT_EQ(msg_members->size(), pg->pg_info_.members.size());
+    for (auto m : *msg_members) {
+        uuids::uuid id{};
+        std::copy_n(m->uuid()->data(), 16, id.begin());
+        auto it = pg->pg_info_.members.find(PGMember{id});
+        ASSERT_TRUE(it != pg->pg_info_.members.end());
+        ASSERT_EQ(m->name()->str(), it->name);
+        ASSERT_EQ(m->priority(), it->priority);
+    }
+
+    auto idx = 0;
+    ASSERT_EQ(pg->shards_.size()-1, pg_msg->shard_ids()->size());
+    for (auto& shard : pg->shards_) {
+        if (shard->info.lsn > snp_lsn) { continue; }
+        ASSERT_EQ(shard->info.id, pg_msg->shard_ids()->Get(idx++));
+    }
+
+    //Verify shard meta data
+    auto shard_list = pg_shard_id_vec[1];
+    current_blob_id = 0;
+    for (auto& shard : pg->shards_) {
+        auto shard_seq_num = shard->info.id & 0xFFFFFFFFFFFF;
```

Comment:
Can utilize `get_sequence_num_from_shard_id` if https://github.com/eBay/HomeObject/pull/227 is merged first.

Reply:
Sure, thanks!